src/commands/farm.rs

```diff
@@ -133,72 +133,84 @@ async fn subscribe_to_plotting_progress(
     is_initial_progress_finished: Arc<AtomicBool>,
     sector_size_bytes: u64,
 ) {
-    for (plot_id, plot) in farmer.iter_plots().await.enumerate() {
-        println!("Initial plotting for plot: #{plot_id} ({})", plot.directory().display());
-
-        let mut plotting_progress = plot.subscribe_initial_plotting_progress().await;
-
-        if let Some(plotting_result) = plotting_progress.next().await {
-            let current_size = plotting_result.current_sector * sector_size_bytes;
-            let progress_bar = plotting_progress_bar(current_size, plot.allocated_space().as_u64());
-
-            while let Some(stream_result) = plotting_progress.next().await {
-                let current_size = stream_result.current_sector * sector_size_bytes;
-                progress_bar.set_position(current_size);
+    tokio::spawn(async move {
```

Comment:
It is better to extract this `tokio::spawn` outside of the function. You can use: `tokio::spawn(subscribe_to_plotting_progress(...))`

Reply:
much better, thanks!